File: learning/ojayer/Demo/checkstorage.js
const { google } = require('googleapis');
const fs = require('fs');
const readline = require('readline');
const path = require('path');

// Load or create .env file
const envPath = path.join(__dirname, '.env');
if (!fs.existsSync(envPath)) {
  fs.writeFileSync(envPath, ''); // Create empty .env if missing
}
const credentials = JSON.parse(fs.readFileSync('credentials.json', 'utf8')).web;
// Configure OAuth2 client
const oauth2Client = new google.auth.OAuth2(
    credentials.client_id,
    credentials.client_secret,
    credentials.redirect_uris[0],
);

const SCOPES = ['https://www.googleapis.com/auth/drive.metadata.readonly'];
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});


// Check storage using saved tokens
async function checkAllStorage() {
    require('dotenv').config(); // Load .env tokens
    const accounts = [
        { name: 'Account 1', refresh_token: process.env.ACCOUNT_1_REFRESH_TOKEN },
        { name: 'Account 2', refresh_token: process.env.ACCOUNT_2_REFRESH_TOKEN },
        { name: 'Account 3', refresh_token: process.env.ACCOUNT_3_REFRESH_TOKEN },
    ];

    let total_Storage = 0;
    let total_Used = 0;
    for (const account of accounts) {
    try {
      oauth2Client.setCredentials({ refresh_token: account.refresh_token });
      const drive = google.drive({ version: 'v3', auth: oauth2Client });
      const res = await drive.about.get({ fields: 'storageQuota' });

      const limit = parseInt(res.data.storageQuota.limit);
      const usage = parseInt(res.data.storageQuota.usage);
      total_Storage += limit;
      total_Used += usage;
      console.log(`\n--- ${account.name} ---`);
      console.log(`Total: ${Math.round(limit / (1024**3))} GB`);
      console.log(`Used: ${Math.round(usage / (1024**3))} GB`);
    } catch (error) {
      console.error(`Error for ${account.name}:`, error.message);
    }
  }
    console.log(`\n--- Storage details ---`);
    console.log(`Total Storage: ${Math.round(total_Storage / (1024**3))} GB`);
    console.log(`Total Used: ${Math.round(total_Used / (1024**3))} GB`);
    console.log(`Total Free: ${Math.round((total_Storage - total_Used) / (1024**3))} GB`);

    return;
}

// Start authorization for Account 1
//authorizeAndSaveToken(1);
checkAllStorage().catch(console.error);

File: learning/ojayer/Py_Demo1/checkstorage.py
import os
import json
from google.oauth2.credentials import Credentials
from googleapiclient.discovery import build
from google_auth_oauthlib.flow import InstalledAppFlow
from dotenv import load_dotenv

# Load or create .env file
ENV_PATH = '.env'
if not os.path.exists(ENV_PATH):
    with open(ENV_PATH, 'w') as f:
        f.write('')

load_dotenv()  # Load environment variables

# Load credentials
with open('credentials.json', 'r') as f:
    credentials = json.load(f)['installed']

SCOPES = ['https://www.googleapis.com/auth/drive.metadata.readonly']
TOTAL_ACCOUNTS = 3


def check_storage(account):
    try:
            creds = Credentials(
                None,  # Access token (not needed, as we use refresh token)
                refresh_token=account['refresh_token'],
                client_id=credentials['client_id'],
                client_secret=credentials['client_secret'],
                token_uri='https://oauth2.googleapis.com/token'
            )
            drive_service = build('drive', 'v3', credentials=creds)
            res = drive_service.about().get(fields='storageQuota').execute()
            
            limit = int(res['storageQuota']['limit'])
            usage = int(res['storageQuota']['usage'])
            
            
            print(f"\n--- {account['name']} ---")
            print(f"Total: {round(limit / (1024**3))} GB")
            print(f"Used: {round(usage / (1024**3))} GB")
            
    except Exception as e:
            print(f"Error for {account['name']}: {e}")
    
    return limit,usage

def check_all_storage():
    accounts = [
        {"name": "Account 1", "refresh_token": os.getenv("ACCOUNT_1_REFRESH_TOKEN")},
        {"name": "Account 2", "refresh_token": os.getenv("ACCOUNT_2_REFRESH_TOKEN")},
        {"name": "Account 3", "refresh_token": os.getenv("ACCOUNT_3_REFRESH_TOKEN")},
    ]
    
    total_storage = 0
    total_used = 0
    
    for account in accounts:
        storage,used = check_storage(account)
        total_storage+=storage
        total_used+=used
    print("\n--- Storage details ---")
    print(f"Total Storage: {round(total_storage / (1024**3))} GB")
    print(f"Total Used: {round(total_used / (1024**3))} GB")
    print(f"Total Free: {round((total_storage - total_used) / (1024**3))} GB")

if __name__ == "__main__":
    check_all_storage()
